test/dates/io.jl

```diff
@@ -267,8 +291,7 @@ f = "y m d"
 @test_throws ArgumentError Dates.Date("1 1 32",f)
 @test_throws ArgumentError Dates.Date(" 1 1 32",f)
 @test_throws ArgumentError Dates.Date("# 1 1 32",f)
-# can't find 1st space delimiter,s o fails
-@test_throws ArgumentError Dates.Date("1",f)
+@test Dates.Date("1",f) == Dates.Date(1)
```

Comment:
There was a special case in the original code that said: if the first slot is succeeded by a delimiter then the first slot and the delimiter is required.

Before PR
```julia
julia> Dates.Date("1", "y-m-d")
ERROR: ArgumentError: Delimiter mismatch. Couldn't find first delimiter, "-", in date string
 in parse(::String, ::Base.Dates.DateFormat) at ./dates/io.jl:169
 in Date(::String, ::String) at ./dates/io.jl:277

julia> Dates.Date("1-", "y-m-d")
0001-01-01

julia>  Dates.Date("1", "ymd")
0001-01-01
```

After PR:
```julia
julia> Dates.Date("1", "y-m-d")
0001-01-01

julia> Dates.Date("1-", "y-m-d")
0001-01-01

julia> Dates.Date("1", "ymd")  # Note: currently there is a bug with this.
0001-01-01
```